src/commands/utils/download.cc

```diff
@@ -235,7 +235,8 @@ int DownloadCmd::execute( Zypper &zypper , const std::vector<std::string> &posit
       if ( q.empty() || !isPackageType( *q.begin() ) )
       {
         // translators: Label text; is followed by ': cmdline argument'
-        zypper.out().warning( str::Str() << _("Argument resolves to no package") << ": " << pkgspec.orig_str );
+        zypper.out().error( str::Str() << _("Argument resolves to no package") << ": " << pkgspec.orig_str );
+        zypper.setExitCode( ZYPPER_EXIT_INF_CAP_NOT_FOUND );
```

Comment:
```suggestion
        zypper.setExitInfoCode( ZYPPER_EXIT_INF_CAP_NOT_FOUND );
```
(and revert the two return statements below)

Your idea is basically right. The ExitInfo code however is returned IFF zypper would have exited OK. This way you're not in danger to accidentally overwrite some 'more important' error code which has been set before.